Python/traceback.c

```diff
@@ -501,8 +511,35 @@ _Py_DisplaySourceLine(PyObject *f, PyObject *filename, int lineno, int indent)
     return err;
 }
 
+static int byte_to_character_offset_in_line(PyObject* line, int offset) {
+    // Taken pretty much exactly from pegen.c for now.
+    if (offset <= 0) {
+        return offset;
+    }
+    const char *str = PyUnicode_AsUTF8(line);
+    if (!str) {
+        return 0;
+    }
+    Py_ssize_t len = strlen(str);
+    if (offset > len) {
+        offset = Py_SAFE_DOWNCAST(len, Py_ssize_t, int);
+    }
+
+    PyObject *text = PyUnicode_DecodeUTF8(str, offset, "replace");
+    if (!text) {
+        return 0;
+    }
+    Py_ssize_t size = PyUnicode_GET_LENGTH(text);
+    Py_DECREF(text);
+    return Py_SAFE_DOWNCAST(size, Py_ssize_t, int);
+}
+
+#define _TRACEBACK_SOURCE_LINE_INDENT 4
+
+// TODO: Pick up filename and other stuff from the tb argument
```

Comment:
We should probably fix this todo here, no?

Reply:
@isidentical I think you added in this `TODO`, is this still something we should do?

Reply:
I don't recall adding this, feel free to drop it